### ISSUE TITLE
when i use YOLOX to train on my VOC dataset,the model didn't converge

### ISSUE THREAD

**Issue by zhengmingzhang** [open-mmlab/mmdetection#5940]
my config is:
~~~ python
Config:
optimizer = dict(
    type='SGD',
    lr=0.01,
    momentum=0.9,
    weight_decay=0.0005,
    nesterov=True,
    paramwise_cfg=dict(norm_decay_mult=0.0, bias_decay_mult=0.0))
optimizer_config = dict(grad_clip=None)
lr_config = dict(
    policy='YOLOX',
    warmup='exp',
    by_epoch=False,
    warmup_by_epoch=True,
    warmup_ratio=1,
    warmup_iters=5,
    num_last_epochs=15,
    min_lr_ratio=0.05)
runner = dict(type='EpochBasedRunner', max_epochs=300)
checkpoint_config = dict(interval=12)
log_config = dict(interval=50, hooks=[dict(type='TextLoggerHook')])
custom_hooks = [
    dict(type='YOLOXModeSwitchHook', num_last_epochs=15, priority=48),
    dict(
        type='SyncRandomSizeHook',
        ratio_range=(14, 26),
        img_scale=(640, 640),
        interval=10,
        priority=48),
    dict(type='SyncNormHook', num_last_epochs=15, interval=10, priority=48),
    dict(type='ExpMomentumEMAHook', resume_from=None, priority=49)
]
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = None
resume_from = None
workflow = [('train', 1)]
model = dict(
    type='YOLOX',
    backbone=dict(type='CSPDarknet', deepen_factor=0.33, widen_factor=0.5),
    neck=dict(
        type='YOLOXPAFPN',
        in_channels=[128, 256, 512],
        out_channels=128,
        num_csp_blocks=1),
    bbox_head=dict(
        type='YOLOXHead', num_classes=10, in_channels=128, feat_channels=128),
    train_cfg=dict(assigner=dict(type='SimOTAAssigner', center_radius=2.5)),
    test_cfg=dict(score_thr=0.01, nms=dict(type='nms', iou_threshold=0.65)))
data_root = './xray/VOCdevkit/'
dataset_type = 'XRAYDataset'
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
img_scale = (640, 640)
train_pipeline = [
    dict(type='Mosaic', img_scale=(640, 640), pad_val=114.0),
    dict(
        type='RandomAffine', scaling_ratio_range=(0.1, 2),
        border=(-320, -320)),
    dict(
        type='MixUp',
        img_scale=(640, 640),
        ratio_range=(0.8, 1.6),
        pad_val=114.0),
    dict(
        type='PhotoMetricDistortion',
        brightness_delta=32,
        contrast_range=(0.5, 1.5),
        saturation_range=(0.5, 1.5),
        hue_delta=18),
    dict(type='RandomFlip', flip_ratio=0.5),
    dict(type='Resize', keep_ratio=True),
    dict(type='Pad', pad_to_square=True, pad_val=114.0),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_rgb=True),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
]
train_dataset = dict(
    type='MultiImageMixDataset',
    dataset=dict(
        type='XRAYDataset',
        ann_file='./xray/VOCdevkit/VOC2007/ImageSets/Main/train.txt',
        img_prefix='./xray/VOCdevkit/VOC2007/',
        pipeline=[
            dict(type='LoadImageFromFile', to_float32=True),
            dict(type='LoadAnnotations', with_bbox=True)
        ],
        filter_empty_gt=False),
    pipeline=[
        dict(type='Mosaic', img_scale=(640, 640), pad_val=114.0),
        dict(
            type='RandomAffine',
            scaling_ratio_range=(0.1, 2),
            border=(-320, -320)),
        dict(
            type='MixUp',
            img_scale=(640, 640),
            ratio_range=(0.8, 1.6),
            pad_val=114.0),
        dict(
            type='PhotoMetricDistortion',
            brightness_delta=32,
            contrast_range=(0.5, 1.5),
            saturation_range=(0.5, 1.5),
            hue_delta=18),
        dict(type='RandomFlip', flip_ratio=0.5),
        dict(type='Resize', keep_ratio=True),
        dict(type='Pad', pad_to_square=True, pad_val=114.0),
        dict(
            type='Normalize',
            mean=[123.675, 116.28, 103.53],
            std=[58.395, 57.12, 57.375],
            to_rgb=True),
        dict(type='DefaultFormatBundle'),
        dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
    ],
    dynamic_scale=(640, 640))
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(640, 640),
        flip=False,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(type='Pad', size=(640, 640), pad_val=114.0),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='DefaultFormatBundle'),
            dict(type='Collect', keys=['img'])
        ])
]
data = dict(
    samples_per_gpu=16,
    workers_per_gpu=1,
    train=dict(
        type='MultiImageMixDataset',
        dataset=dict(
            type='XRAYDataset',
            ann_file='./xray/VOCdevkit/VOC2007/ImageSets/Main/test.txt',
            img_prefix='./xray/VOCdevkit/VOC2007/',
            pipeline=[
                dict(type='LoadImageFromFile', to_float32=True),
                dict(type='LoadAnnotations', with_bbox=True)
            ],
            filter_empty_gt=False),
        pipeline=[
            dict(type='Mosaic', img_scale=(640, 640), pad_val=114.0),
            dict(
                type='RandomAffine',
                scaling_ratio_range=(0.1, 2),
                border=(-320, -320)),
            dict(
                type='MixUp',
                img_scale=(640, 640),
                ratio_range=(0.8, 1.6),
                pad_val=114.0),
            dict(
                type='PhotoMetricDistortion',
                brightness_delta=32,
                contrast_range=(0.5, 1.5),
                saturation_range=(0.5, 1.5),
                hue_delta=18),
            dict(type='RandomFlip', flip_ratio=0.5),
            dict(type='Resize', keep_ratio=True),
            dict(type='Pad', pad_to_square=True, pad_val=114.0),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='DefaultFormatBundle'),
            dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
        ],
        dynamic_scale=(640, 640)),
    val=dict(
        type='XRAYDataset',
        ann_file='./xray/VOCdevkit/VOC2007/ImageSets/Main/val.txt',
        img_prefix='./xray/VOCdevkit/VOC2007/',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(640, 640),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(type='Pad', size=(640, 640), pad_val=114.0),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='DefaultFormatBundle'),
                    dict(type='Collect', keys=['img'])
                ])
        ]),
    test=dict(
        type='XRAYDataset',
        ann_file='./xray/VOCdevkit/VOC2007/ImageSets/Main/test.txt',
        img_prefix='./xray/VOCdevkit/VOC2007/',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(640, 640),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(type='Pad', size=(640, 640), pad_val=114.0),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='DefaultFormatBundle'),
                    dict(type='Collect', keys=['img'])
                ])
        ]))
interval = 10
evaluation = dict(interval=12, metric='mAP')
work_dir = './tutorial_exps'
seed = 0
gpu_ids = range(0, 1)
~~~

I trained 300 epochs, the MAP on val is always 0

**Comment by hhaAndroid** [comment]
@zhengmingzhang Thank you for your feedback. Can you post the log? The above lr=0.01 is set based on 64 bs, if you are 16 bs, please set it to 0.01/4.

**Comment by zhengmingzhang** [comment]
@hhaAndroid 
Thank you for your reply, there is my log,i tried 0.01/8,batch_size is 8.there must be something wrong here
~~~python
{"mode": "val", "epoch": 10, "iter": 19, "lr": 0.00125, "AP50": 0.0, "mAP": 0.0}
{"mode": "val", "epoch": 20, "iter": 19, "lr": 0.00124, "AP50": 0.0, "mAP": 0.0}
{"mode": "val", "epoch": 30, "iter": 19, "lr": 0.00123, "AP50": 0.0, "mAP": 0.0}
{"mode": "val", "epoch": 40, "iter": 19, "lr": 0.0012, "AP50": 0.0, "mAP": 0.0}
{"mode": "val", "epoch": 50, "iter": 19, "lr": 0.00118, "AP50": 0.0, "mAP": 6e-05}
{"mode": "val", "epoch": 60, "iter": 19, "lr": 0.00114, "AP50": 0.0, "mAP": 8e-05}
{"mode": "val", "epoch": 70, "iter": 19, "lr": 0.0011, "AP50": 0.002, "mAP": 0.00186}
{"mode": "val", "epoch": 80, "iter": 19, "lr": 0.00105, "AP50": 0.055, "mAP": 0.05455}
{"mode": "val", "epoch": 90, "iter": 19, "lr": 0.001, "AP50": 0.002, "mAP": 0.00227}
{"mode": "val", "epoch": 100, "iter": 19, "lr": 0.00094, "AP50": 0.0, "mAP": 0.00048}
{"mode": "val", "epoch": 110, "iter": 19, "lr": 0.00088, "AP50": 0.0, "mAP": 0.00018}
{"mode": "val", "epoch": 120, "iter": 19, "lr": 0.00082, "AP50": 0.0, "mAP": 0.00017}
{"mode": "val", "epoch": 130, "iter": 19, "lr": 0.00076, "AP50": 0.0, "mAP": 0.00019}
{"mode": "val", "epoch": 140, "iter": 19, "lr": 0.00069, "AP50": 0.0, "mAP": 0.0002}
{"mode": "val", "epoch": 150, "iter": 19, "lr": 0.00062, "AP50": 0.0, "mAP": 0.00018}
{"mode": "val", "epoch": 160, "iter": 19, "lr": 0.00056, "AP50": 0.0, "mAP": 0.00042}
{"mode": "val", "epoch": 170, "iter": 19, "lr": 0.00049, "AP50": 0.0, "mAP": 0.00034}
{"mode": "val", "epoch": 180, "iter": 19, "lr": 0.00043, "AP50": 0.0, "mAP": 0.00043}
{"mode": "val", "epoch": 190, "iter": 19, "lr": 0.00037, "AP50": 0.0, "mAP": 0.00024}
{"mode": "val", "epoch": 200, "iter": 19, "lr": 0.00031, "AP50": 0.01, "mAP": 0.00999}
~~~

**Comment by zhengmingzhang** [comment]
> @zhengmingzhang Thank you for your feedback. Can you post the log? The above lr=0.01 is set based on 64 bs, if you are 16 bs, please set it to 0.01/4.

and the log even no loss,i don't know where i did wrong

**Comment by hhaAndroid** [comment]
> > @zhengmingzhang Thank you for your feedback. Can you post the log? The above lr=0.01 is set based on 64 bs, if you are 16 bs, please set it to 0.01/4.
> 
> and the log even no loss,i don't know where i did wrong

The default printing interval is 50. If your dataset length is less than 50xbs, then it will not be printed. You can consider modifying this parameter.

**Comment by zhengmingzhang** [comment]
> > > @zhengmingzhang Thank you for your feedback. Can you post the log? The above lr=0.01 is set based on 64 bs, if you are 16 bs, please set it to 0.01/4.
> > 
> > 
> > and the log even no loss,i don't know where i did wrong
> 
> The default printing interval is 50. If your dataset length is less than 50xbs, then it will not be printed. You can consider modifying this parameter.

but my dataset length is 3600,maybe the method i load the dataset is wrong?
